lib/backpex/fields/has_many.ex

```diff
@@ -293,40 +231,71 @@ defmodule Backpex.Fields.HasMany do
     {:noreply, socket}
   end
 
-  @impl Phoenix.LiveComponent
-  def handle_event("toggle-select-all", _params, socket) do
-    %{
-      assigns:
-        %{name: name, field_options: field_options, show_select_all: show_select_all, options_count: options_count} =
-          assigns
-    } = socket
+  @impl Backpex.Field
+  def display_field({_name, field_options}), do: Map.get(field_options, :display_field)
 
-    socket =
-      if show_select_all do
-        options = options(assigns)
+  @impl Backpex.Field
+  def association?(_field), do: true
 
-        put_assoc(name, options)
+  @impl Backpex.Field
+  def schema({name, _field_options}, schema) do
+    schema.__schema__(:association, name)
+    |> Map.get(:queryable)
+  end
 
-        socket
-        |> assign(:options, options)
-        |> assign(:selected, options)
-        |> assign(:show_more, false)
-      else
-        options = options(assigns, limit: query_limit(field_options))
+  @impl Backpex.Field
+  def before_changeset(changeset, attrs, _metadata, repo, field, assigns) do
+    {field_name, field_options} = field
+    validate_live_resource(field_name, field_options)
```

Comment:
Things like this should be done way more generic and for every field / filter / etc. (in the future).

https://github.com/dashbitco/nimble_options

Maybe even on compile time.